gapic/src/main/com/google/gapid/views/DeviceDialog.java

```diff
@@ -0,0 +1,322 @@
+        return composite;
+      }
+
+      // Recap capture info
+      createLabel(composite, "Capture name: " + models.capture.getName());
+      createLabel(composite,
+          "Capture device: " + Devices.getLabel(models.capture.getData().capture.getDevice()));
+
+      // Warning when no compatible device found
+      noCompatibleDeviceFound = createLabel(composite, Messages.SELECT_DEVICE_NO_COMPATIBLE_FOUND);
+      noCompatibleDeviceFound.setForeground(theme.missingInput());
+
+      // Mirror the device combo from TracerDialog
+      Group mainGroup =
+          withLayoutData(createGroup(composite, "Select replay device", new GridLayout(3, false)),
+              new GridData(GridData.FILL_HORIZONTAL));
+      createLabel(mainGroup, "Device:");
+      deviceCombo = createDropDownViewer(mainGroup);
+      deviceCombo.setContentProvider(ArrayContentProvider.getInstance());
+      deviceCombo.getCombo().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+      deviceCombo.setLabelProvider(new LabelProvider() {
+        @Override
+        public String getText(Object element) {
+          return Devices.getLabel(((Device.Instance) element));
+        }
+      });
+
+      deviceLoader = widgets.loading.createWidgetWithRefresh(mainGroup);
+      deviceLoader
+          .setLayoutData(withIndents(new GridData(SWT.RIGHT, SWT.CENTER, false, false), 5, 0));
+      // TODO: Make this a true button to allow keyboard use.
+      deviceLoader.addListener(SWT.MouseDown, e -> {
+        deviceLoader.startLoading();
+        // By waiting a tiny bit, the icon will change to the loading indicator, giving the user
+        // feedback that something is happening, in case the refresh is really quick.
+        logFailure(LOG,
+            Scheduler.EXECUTOR.schedule(
+                () -> models.devices.loadReplayDevices(models.capture.getData().path), 300,
+                TimeUnit.MILLISECONDS));
+      });
+
+      validationStatusLoader = widgets.loading.createWidgetWithImage(mainGroup,
+          widgets.theme.check(), widgets.theme.error());
+      validationStatusLoader
+          .setLayoutData(withIndents(new GridData(SWT.LEFT, SWT.BOTTOM, false, false), 0, 0));
+      validationStatusText = createLink(mainGroup, "", e -> {
+        Program.launch(URLs.DEVICE_COMPATIBILITY_URL);
+      });
+      validationStatusText.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+      deviceCombo.getCombo().addListener(SWT.Selection, e -> {
+        runValidationCheck(getSelectedDevice());
+      });
+
+      models.devices.addListener(new Devices.Listener() {
+        @Override
+        public void onReplayDevicesLoaded() {
+          replayDevicesLoaded();
+        }
+      });
+
+      replayDevicesLoaded();
+      return composite;
+    }
+
+    @Override
+    protected void createButtonsForButtonBar(Composite parent) {
+      openTrace = createButton(parent, IDialogConstants.OK_ID, IDialogConstants.OK_LABEL, true);
```

Comment:
You don't need to store the button in an instance field. You can always get it back via `getButton(IDialogConstants.OK_ID)`.

Reply:
Fixed